test/unit/master/test_subjob.py

```diff
@@ -36,6 +36,12 @@ def setUp(self):
             ],
         )
 
+    def test_set_atoms_subjob_id(self):
+        atoms = [Mock(), Mock()]
+        self._subjob._set_atoms_subjob_id(atoms, 4)
+        for atom in atoms:
+            self.assertEqual(atom.subjob_id, 4)
+
```

Comment:
This test is okay but I think we can do better. I think generally we should try to test through public interfaces as much as possible.

It would be better to do a test that passes the list of atoms to the Subjob constructor and then checks that the subjob_id on each atom was set.